client/transaction-pool/graph/src/pool.rs

```diff
@@ -438,6 +447,16 @@ impl<B: ChainApi> Pool<B> {
 		(hash, validity)
 	}
 
+	/// Notify all watchers that transactions in the block with hash been finalized
+	pub async fn finalized(&self, block_hash: BlockHash<B>) -> Result<(), B::Error> {
+		self.validated_pool.finalized(block_hash).await
+	}
+
+	/// Notify the listener of retracted blocks
+	pub fn retracted(&self, block_hash: &BlockHash<B>) {
+		self.validated_pool.retracted(block_hash)
+	}
+
 	/// Get ready transaction by hash, if it present in the pool.
 	pub fn ready_transaction(&self, hash: &ExHash<B>) -> Option<TransactionFor<B>> {
 		self.validated_pool.ready_by_hash(hash)
```

Comment:
I really start to hate the fact that we have to duplicate and delegate so many methods. Perhaps better to just expose `validated_pool` via a getter and call the methods directly?
```
pub fn validated_pool(&self) -> &ValidatedPool {
  &self.validated_pool
}
```
and call via:
```
pool.validated_pool().ready_transaction(hash)
```